pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -2323,52 +2320,52 @@ def buildScriptsAssemble(
+                String testTarget = remoteJobTargets[testIndex]
+                def jobHandle = remoteTriggeredBuilds[testTarget]
+                if ( completedJckJobs.contains(testTarget) ) {
+                    // This job has already finished, so we don't want to see more "waiting" output.'
+                    continue
+                }
+                def remoteJobStatus = ""
+                if ( jobHandle == null ) {
+                    context.println "Failed, remote job ${testTarget} was not triggered"
+                    remoteJobStatus = "FAILURE"
+                } else {
+                    jobHandle.updateBuildStatus()
+                    if ( jobHandle.getBuildStatus().toString().equals("NOT_TRIGGERED") ) {
+                        context.println "Failed, remote job ${testTarget} status is NOT_TRIGGERED"
                         remoteJobStatus = "FAILURE"
                     } else {
-                        jobHandle.updateBuildStatus()
-                        if ( jobHandle.getBuildStatus().toString().equals("NOT_TRIGGERED") ) {
-                            context.println "Failed, remote job ${testTargets} status is NOT_TRIGGERED"
-                            remoteJobStatus = "FAILURE"
+                        if ( !jobHandle.isFinished() ) {
+                            context.println "Current ${testTarget} Status: " + jobHandle.getBuildStatus().toString() + " Remote build URL: " + jobHandle.getBuildUrl();
                         } else {
-                            if ( !jobHandle.isFinished() ) {
-                                waitingForRemoteJck = true
-                                context.println "Current ${testTargets} Status: " + jobHandle.getBuildStatus().toString() + " Remote build URL: " + jobHandle.getBuildUrl();
-                            } else {
-                                remoteJobStatus = jobHandle.getBuildResult().toString()
-                                context.println "Current ${testTargets} Status: " + jobHandle.getBuildStatus().toString() + " Build status: ${remoteJobStatus}" + " Remote build URL: " + jobHandle.getBuildUrl();
-                            }
+                            remoteJobStatus = jobHandle.getBuildResult().toString()
+                            context.println "Current ${testTarget} Status: " + jobHandle.getBuildStatus().toString() + " Build status: ${remoteJobStatus}" + " Remote build URL: " + jobHandle.getBuildUrl();
                         }
                     }
-                    if ( remoteJobStatus != "" && !completedJckJobs.contains(testTargets)) {
-                        setStageResult("${testTargets}", remoteJobStatus);
-                        completedJckJobs += ",${testTargets}"
+                }
+                if ( remoteJobStatus != "" ) {
+                    context.stage("${testTarget}") {
+                        setStageResult("${testTarget}", remoteJobStatus)
```

Comment:
This remotejob status stage can move to when all jobs finish, i.e, remoteTriggeredBuilds.size() == completedJckJobCount.

Reply:
Hi Sophia. I'm not sure why you'd want this, as that means we won't get a stage for sanity.jck's results (which may complete quickly) until after extended.jck has finished (which could take much longer).